app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,46 @@
+class MerchantsController < ApplicationController
+  def index
+    @merchants = Merchant.all
+  end
+
+  def new
+  end
+
+  def create
+    merchant = Merchant.new({
+      name: params[:name],
+      address: params[:address],
+      city: params[:city],
+      state: params[:state],
+      zip: params[:zip],
+      })
+      merchant.save
+      redirect_to '/merchants'
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+  end
+
+  def edit
+    show
+    # @merchant = Merchant.find(params[:id])
+  end
+
+  def update
+    merchant = Merchant.find(params[:id])
+    merchant.update(merchant_params)
+    redirect_to "/merchants/#{merchant.id}"
+  end
+
+  def destroy
+    @merchant = Merchant.find(params[:id])
+    @merchant.destroy
+    redirect_to '/merchants'
+  end
```

Comment:
This destroy method will work only if your merchants don't have any items. You need to delete any items associated with that merchant first, and then you can delete the merchant.